src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -20701,19 +20731,67 @@ public static bool TryGetValue([MaybeNullWhen(true)] out string s)
         if (b)
         {
             s = null;
-            return false;
+            return false; // 1
         }
         else
         {
             s = """";
             return true;
         }
     }
+
+    public static bool TryGetValue2([MaybeNullWhen(true)] out string s)
+    {
+        bool b = true;
+        if (b)
+        {
+            s = """";
+            return false;
+        }
+        else
+        {
+            s = null;
+            return true;
+        }
+    }
 }
 ";
-            // Since we can't analyze this well enough, we loosen the type unconditionally
             var comp = CreateNullableCompilation(new[] { source, MaybeNullWhenAttributeDefinition });
-            comp.VerifyDiagnostics();
+            comp.VerifyDiagnostics(
+                // (11,13): warning CS8762: Parameter 's' may not have a null value when exiting with 'false'.
+                //             return false; // 1
+                Diagnostic(ErrorCode.WRN_ParameterConditionallyDisallowsNull, "return false;").WithArguments("s", "false").WithLocation(11, 13)
+                );
+        }
```

Comment:
>} [](start = 8, length = 1)

Please test:
```C#
static bool TryGetValue([MaybeNullWhen(false)]out string s)
{
    s = null;
    return (bool)true;
}
```
It doesn't necessarily need to warn but we should test the behavior.

Reply:
Added the test
The cast ruins the conditional analysis (ie. no warning on `return (bool)true;` nor `return (bool)false;`.